### DISCUSSION TITLE
Decimal Strings Instead of Values

### DISCUSSION THREAD

**Discussion by RPeterson0487** [sqlalchemy/sqlalchemy#11228] [Usage Questions] [answered]
Hello,
This is one part of two questions, I decided to split this into separate questions.
I have this code:
```py
cost = database_obj.cost
    inventory = database_obj.inventory
    
    with database_obj.session_factory() as session:
        test = session.execute(
            sql.select(
                cost.Formula,
                cost.Lab_Description,
                sql.func.sum(inventory.Weight),
                sql.func.count("*"),
                inventory.Costs,
                inventory.Purchased
            )
            .join(
                # cost,
                inventory,
                cost.InkCode == inventory.InkType
            )
            .where(
                inventory.Deleted == "N"
            )
            .group_by(
                inventory.Purchased,
                cost.Formula,
                inventory.Costs.desc()
            )
        ).all()
    
    for i in test:
        print(i)
```

Which returns the following results: `('PBAX510378M', 'M0563 REFLEX BLUE MW 5.5', Decimal('35.00'), 1, Decimal('10.260'), 'N')`

My question is this.  Why am I getting "Decimal('35.00') instead of just 35.00?

Thank you much!
-- Rick

**Comment by CaselIT** [comment, marked as answer]
Hi,

That's how python transforms decimal values to strings. 
```py
import decimal
print(repr(decimal.Decimal('35.00')))
# Decimal('35.00')
```

If you want to return a float change the column type on the sqlalchemy table to have `asdecimal=False`

**Reply by RPeterson0487** [reply]
Thank you for getting back to me.
Where would I set that if I reflect the tables?

**Reply by CaselIT** [reply]
You can customize the types using this event. https://docs.sqlalchemy.org/en/20/core/events.html#sqlalchemy.events.DDLEvents.column_reflect

**Reply by gordthompson** [reply]
Note that using floating-point types for monetary values is widely regarded as "wrong". Discussion here:

https://stackoverflow.com/q/3730019/2144390

If you're talking about *displaying* a value as `35.00` instead of `Decimal('35.00')` then that is a formatting issue.

**Reply by RPeterson0487** [reply]
I got with the person who designed the database and he explained why it was decimal.  I appreciate all the help with my questions. :D
